contrib/http_util/src/main/java/io/opencensus/contrib/http/HttpClientHandler.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.contrib.http;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.opencensus.common.ExperimentalApi;
+import io.opencensus.trace.Span;
+import io.opencensus.trace.SpanBuilder;
+import io.opencensus.trace.SpanContext;
+import io.opencensus.trace.Tracer;
+import io.opencensus.trace.propagation.TextFormat;
+import javax.annotation.Nullable;
+
+/**
+ * This helper class provides routine methods to instrument HTTP clients.
+ *
+ * @param <Q> the HTTP request entity.
+ * @param <P> the HTTP response entity.
+ * @param <C> the type of the carrier.
+ * @since 0.13
+ */
+@ExperimentalApi
+public final class HttpClientHandler<Q, P, C> extends HttpHandler<Q, P> {
+
+  private final TextFormat.Setter<C> setter;
+
+  /**
+   * Creates a {@link HttpClientHandler} with given parameters.
+   *
+   * @param tracer the Open Census tracing component.
+   * @param extractor the {@code HttpExtractor} used to extract information from the
+   *     request/response.
+   * @param customizer the {@link HttpSpanCustomizer} used to customize span behaviors.
+   * @param textFormat the {@code TextFormat} used in HTTP propagation.
+   * @param setter the setter used when injecting information to the {@code carrier}.
+   * @since 0.13
+   */
+  public HttpClientHandler(
+      Tracer tracer,
+      HttpExtractor<Q, P> extractor,
+      HttpSpanCustomizer<Q, P> customizer,
```

Comment:
I think in most of the cases people do not want to pass a custom `customizer` since implementation of `HttpSpanCustomizer` already offers most of the things you usually need. I'd suggest to add a second constructor which does not have the `customizer` paramter and uses the current `HttpSpanCustomizer` as default.

Reply:
Personally I think having a component like we do in Brave "HttpTracing" which includes these things is less maintenance and easier to manage. public constructors can be troublesome and this makes it less, moving the varied things to one spot.